raiden_contracts/tests/test_channel_deposit.py

```diff
@@ -1,5 +1,99 @@
+import pytest
+from ethereum import tester
 from raiden_contracts.utils.config import E_CHANNEL_NEW_DEPOSIT
-from .utils import check_new_deposit
+from raiden_contracts.utils.events import check_new_deposit
+from .fixtures.config import (
+    empty_address,
+    fake_address,
+)
+
+
+def test_deposit_channel_call(token_network, custom_token, create_channel, get_accounts):
+    (A, B) = get_accounts(2)
+    channel_identifier = create_channel(A, B)
+
+    custom_token.transact({'from': A, 'value': 10 ** 18}).mint()
+    deposit_A = custom_token.call().balanceOf(A)
+
+    custom_token.transact({'from': A}).approve(token_network.address, deposit_A)
+
+    with pytest.raises(TypeError):
+        token_network.transact({'from': A}).setDeposit(
+            -1,
+            A,
+            deposit_A
+        )
+    with pytest.raises(TypeError):
+        token_network.transact({'from': A}).setDeposit(
+            channel_identifier,
+            '',
+            deposit_A
+        )
+    with pytest.raises(TypeError):
+        token_network.transact({'from': A}).setDeposit(
+            channel_identifier,
+            fake_address,
+            deposit_A
+        )
+    with pytest.raises(TypeError):
+        token_network.transact({'from': A}).setDeposit(
+            channel_identifier,
+            0x0,
+            deposit_A
+        )
+    with pytest.raises(TypeError):
```

Comment:
It checks the types used by the contract's functions to be something that is expected.
E.g. in which `TypeErrors` are raised: 
- negative `int` instead of `uint` 
- non-address like `string` instead of an `address` etc.